main.go

```diff
@@ -67,15 +73,26 @@ func main() {
 	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, time.Second*30)
 	diffsnapInformerFactory := informers.NewSharedInformerFactory(diffsnapClient, time.Second*30)
 
-	getChangedBlocksesInformer := informer.New(diffsnapInformerFactory, CONTROLLER_NAMESPACE, nil).GetChangedBlockses()
-	controller := NewController(kubeClient, diffsnapClient, getChangedBlocksesInformer)
-		//diffsnapInformerFactory.Differentialsnapshot().V1alpha1().GetChangedBlockses())
+	metricsManager := metrics.NewCSIMetricsManagerForSidecar("cbt-service")
+
+	klog.Infof("csi address: %v", csiAddress)
 
-	if controller == nil {
-		klog.Fatalf("Error initialize controller.")
-		return
+	//create client
+	csiConn, err := connection.Connect(
+		csiAddress,
+		metricsManager,
+		connection.OnConnectionLoss(connection.ExitOnConnectionLoss()))
+	if err != nil {
+		klog.Errorf("error connecting to CSI driver: %v", err)
+		os.Exit(1)
 	}
 
+	cbtClient := changeblockservice.NewDifferentialSnapshotClient(csiConn)
+
+	controller := controller.NewController(kubeClient, exampleClient,
+		exampleInformerFactory.Differentialsnapshot().V1alpha1().GetChangedBlockses(),
```

Comment:
This line won't work because the namespace is not specified so it will be treated as cluster scope while our GetChangedBlocks CRD is namespace scoped.

Reply:
I don't think that is correct, why do we want to limit the controller to only watch in a single namespace?

A cluster scoped watch of namespaced objects does work and just means that getchangedblocks created in namespace a where the volume snapshots are we will get that in the informer. Can you run the code to make sure as this did work for me

Reply:
Let's just go ahead and merge in.  Then we will see if it needs to be adjusted afterward.